// @quasar/quasar-ui-qflashcard/dist/index.umd.js
/*!
 * @quasar/quasar-ui-qflashcard v1.0.5
 * (c) 2020 Jeff Galbraith <jeff@quasar.dev>
 * Released under the MIT License.
 */

(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('vue')) :
  typeof define === 'function' && define.amd ? define(['exports', 'vue'], factory) :
  (global = global || self, factory(global.QFlashcard = {}, global.Vue));
}(this, (function (exports, Vue) { 'use strict';

  Vue = Vue && Object.prototype.hasOwnProperty.call(Vue, 'default') ? Vue['default'] : Vue;

  var version = "1.0.5";

  var QFlashcard = {
    name: 'QFlashcard',

    props: {
      noHover: Boolean
    },

    render: function render (h) {
      var slot = this.$slots.default;

      return h('div', {
        staticClass: 'q-flashcard' + (this.noHover === true ? ' no-hover' : '')
      }, slot)
    }
  };

  var QFlashcardSection = {
    name: 'QFlashcardSection',

    props: {
      active: Boolean,
      transition: [String, Array]
    },

    computed: {
      classes: function classes () {
        var this$1 = this;

        if (this.transition === void 0) {
          return ''
        }
        var transition = this.transition;
        if (typeof transition === 'string') {
          if (transition.includes(' ')) {
            // now transition is an array and handled below...
            transition = this.transition.split(' ');
          } else {
            return this.__transitionName(transition)
          }
        }

        if (Array.isArray(transition)) {
          return transition
            .map(function (t) { return this$1.__transitionName(t); })
            .join(' ')
        }

        return ''
      }
    },

    methods: {
      __transitionName: function __transitionName (transition) {
        var postfix = this.active === true ? '--active' : '';
        return (transition.startsWith('fc-')
          ? transition
          : 'fc-' + transition) + postfix
      }
    },

    render: function render (h) {
      var slot = this.$slots.default;

      return h('div', {
        staticClass: 'q-flashcard__section',
        class: this.classes
      }, slot)
    }
  };

  var Plugin = {
    version: version,
    QFlashcard: QFlashcard,
    QFlashcardSection: QFlashcardSection,

    install: function install (Vue) {
      Vue.component(QFlashcard.name, QFlashcard);
      Vue.component(QFlashcardSection.name, QFlashcardSection);
    }
  };

  Vue.use(Plugin);

  exports.QFlashcard = QFlashcard;
  exports.QFlashcardSection = QFlashcardSection;
  exports.version = version;

  Object.defineProperty(exports, '__esModule', { value: true });

})));
